From 81e5647c8a395364eb72d042a9b14f2c59bc3ab9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Syrj=C3=A4l=C3=A4?= <ville.syrjala@linux.intel.com>
Date: Wed, 4 Oct 2023 18:55:58 +0300
Subject: drm/i915: Constify the crtc states in the DPLL checker
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Git-commit: cfc7109bf6d1f51548308beb52326035304313cb
Patch-mainline: v6.7-rc1
References: jsc#PED-3527 jsc#PED-5475 jsc#PED-6068 jsc#PED-6070 jsc#PED-6116 jsc#PED-6120 jsc#PED-5065 jsc#PED-5477 jsc#PED-5511 jsc#PED-6041 jsc#PED-6069 jsc#PED-6071

The DPLL state checker should not be modifying the crtc states,
so make the const.

Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20231004155607.7719-4-ville.syrjala@linux.intel.com
Reviewed-by: Jani Nikula <jani.nikula@intel.com>
Acked-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/gpu/drm/i915/display/intel_dpll_mgr.c | 6 +++---
 drivers/gpu/drm/i915/display/intel_dpll_mgr.h | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/display/intel_dpll_mgr.c b/drivers/gpu/drm/i915/display/intel_dpll_mgr.c
index 86cdd8c9f2d8..237cfc8780a4 100644
--- a/drivers/gpu/drm/i915/display/intel_dpll_mgr.c
+++ b/drivers/gpu/drm/i915/display/intel_dpll_mgr.c
@@ -4460,7 +4460,7 @@ static void
 verify_single_dpll_state(struct drm_i915_private *i915,
 			 struct intel_shared_dpll *pll,
 			 struct intel_crtc *crtc,
-			 struct intel_crtc_state *new_crtc_state)
+			 const struct intel_crtc_state *new_crtc_state)
 {
 	struct intel_dpll_hw_state dpll_hw_state;
 	u8 pipe_mask;
@@ -4513,8 +4513,8 @@ verify_single_dpll_state(struct drm_i915_private *i915,
 }
 
 void intel_shared_dpll_state_verify(struct intel_crtc *crtc,
-				    struct intel_crtc_state *old_crtc_state,
-				    struct intel_crtc_state *new_crtc_state)
+				    const struct intel_crtc_state *old_crtc_state,
+				    const struct intel_crtc_state *new_crtc_state)
 {
 	struct drm_i915_private *i915 = to_i915(crtc->base.dev);
 
diff --git a/drivers/gpu/drm/i915/display/intel_dpll_mgr.h b/drivers/gpu/drm/i915/display/intel_dpll_mgr.h
index 75d2ff977b4e..e184680606e9 100644
--- a/drivers/gpu/drm/i915/display/intel_dpll_mgr.h
+++ b/drivers/gpu/drm/i915/display/intel_dpll_mgr.h
@@ -370,8 +370,8 @@ enum intel_dpll_id icl_tc_port_to_pll_id(enum tc_port tc_port);
 bool intel_dpll_is_combophy(enum intel_dpll_id id);
 
 void intel_shared_dpll_state_verify(struct intel_crtc *crtc,
-				    struct intel_crtc_state *old_crtc_state,
-				    struct intel_crtc_state *new_crtc_state);
+				    const struct intel_crtc_state *old_crtc_state,
+				    const struct intel_crtc_state *new_crtc_state);
 void intel_shared_dpll_verify_disabled(struct drm_i915_private *i915);
 
 #endif /* _INTEL_DPLL_MGR_H_ */
-- 
2.43.0

